### PR TITLE
Add `Duration::MAX` associated constant

### PR DESCRIPTION
**Pull request by marmeladema**
(To merge after https://github.com/rust-lang/rust/pull/72434)

This change introduce `Duration::MAX` as associated constant on the same model as `u64::MAX` etc.

I re-used `const_checked_int_methods` feature gate to constify `Duration::new`, it might not be the best approach but i thought this PR would help start the discussion.

### CODE REVIEW

File: src/libcore/time.rs [hunk 1]
```diff
@@ -130,10 +143,12 @@ impl Duration {
     /// ```
     #[stable(feature = "duration", since = "1.3.0")]
     #[inline]
-    #[rustc_const_stable(feature = "duration_consts", since = "1.32.0")]
-    pub fn new(secs: u64, nanos: u32) -> Duration {
-        let secs =
-            secs.checked_add((nanos / NANOS_PER_SEC) as u64).expect("overflow in Duration::new");
+    #[rustc_const_unstable(feature = "const_checked_int_methods", issue = "53718")]
```
Comment (lcnr) [inline src/libcore/time.rs:146]:
```suggestion
    #[rustc_const_unstable(feature = "const_duration_new", issue = "53718")]
```
I don't think we should reuse `const_checked_int_methods` here.
You might have to add `#![feature(const_duration_new)]` to src/libcore/lib.rs so `Duration::MAX` can use this.

Comment (RalfJung) [reply src/libcore/time.rs:146]: Also there should be a new tracking issue then.

**Comment by RalfJung** [comment]
Not sure I can just approve new APIs like that.^^ Also, there is some overlap with https://github.com/rust-lang/rust/pull/72481.

Cc @rust-lang/wg-const-eval

**Comment by marmeladema** [comment]
Let's close this in favor of #72481 even if it does not introduce `Duration::MAX` quite yet.

**Pull request closed without merging**